app/controllers/admin/custom_fields/hierarchy/items_controller.rb

```diff
@@ -87,6 +87,13 @@ def update
             )
         end
 
+        def move
+          item_service
+            .reorder_item(item: @active_item, new_sort_order: params[:new_sort_order].to_i)
```

Comment:
🟡 `nil.to_i` and `"".to_i` both become 0. Is that intended?

Reply:
I think I handled it better now. Just moved the handling to the service instead of doing magic here. And now requesting the parameter to be present.